examples/machines-with-gce-disks.nix

```diff
@@ -0,0 +1,114 @@
+, instanceServiceAccount ? ""    # Email of the service account to be assigned to the instance
+, instanceScopes ? []            # Access scopes to be assigned to the instance
+, preemptible ? true             # Preemtpible volatile machine
+
+, labels ? {}
+, description ? ""
+, ...
+}:
+{
+  network.description = description;
+
+  defaults =
+    { config, resources, lib, name, uuid, pkgs, ...}:
+    let
+      machineName = "${config.deployment.name}-" + "${name}-" + builtins.substring 2 6 "${uuid}";
+      networkTags = [ "default" ];
+    in
+    rec {
+      deployment.targetEnv = "gce";
+      deployment.gce = {
+        inherit region network subnet serviceAccount machineName;
+        project = gcpProject;
+        tags = networkTags;
+        labels = labels;
+        instanceType = lib.mkDefault instanceType;
+        instanceServiceAccount = {
+          email = instanceServiceAccount;
+          scopes =
+           [
+             "https://www.googleapis.com/auth/devstorage.read_write"
+             "https://www.googleapis.com/auth/logging.write"
+             "https://www.googleapis.com/auth/monitoring.write"
+             "https://www.googleapis.com/auth/trace.append"
+           ] ++ instanceScopes;
+        };
+        accessKey = builtins.readFile accessKey;
+        scheduling = if preemptible then {
+          preemptible = true;
+          automaticRestart = false;
+          onHostMaintenance = "TERMINATE";
+        } else {};
+        rootDiskSize = 100;
+      };
+
+      fileSystems."/data" = {
+        fsType = "xfs";
+        options = [ "noatime" "nodiratime" ];
+        autoFormat = true;
+        formatOptions = "-f";
+        gce.disk = lib.mkDefault null;
+        gce.disk_name = lib.mkDefault null;
+        gce.size = lib.mkDefault volumeSize;
+        gce.diskType = lib.mkDefault "ssd";
+      };
+
+      environment.systemPackages = [ pkgs.google-cloud-sdk ];
+   };
+
+  # GCE Disk for Frontend
+  resources.gceDisks.frontend-volume =
+    { resources, lib, ...}:
+    let
+      namespace = resources.machines.frontend.deployment.gce;
+    in
+    {
+      inherit (namespace) project serviceAccount region accessKey labels;
+      inherit (resources.machines.frontend.fileSystems."/data".gce) size disk diskType;
+      name = "${namespace.machineName}-volume";
+    };
+
+  # Frontend Machine
+  frontend =
+    { resources, lib, ... }:
+    {
+      fileSystems."/data".gce.disk       = resources.gceDisks.frontend-volume;
+      fileSystems."/data".gce.disk_name  = "volume";
+    };
+
+  # GCE Disk for Backend
+  resources.gceDisks.backend-volume =
+    { resources, lib, ...}:
+    let
+      namespace = resources.machines.backend.deployment.gce;
+    in
+    {
+      inherit (namespace) project serviceAccount region accessKey labels;
+      inherit (resources.machines.backend.fileSystems."/data".gce) size disk diskType;
+      name = "${namespace.machineName}-volume";
+      #snapshot ="test";
```

Comment:
avoid commented lines here as well

Reply:
This was added as part of some scenarios testing involving snapshot and image attributes in gce-disk

Eventually, I figured I should keep these in the example file as future reference if anyone wants to create volumes out of a snapshot/image

Do you think I should delete them altogether or move them elsewhere?

Reply:
I think the example file as a whole is a little bit complicated, would be better to make it much simpler as the purpose of examples is just to get started with something simple.

Reply:
Simplified per https://github.com/nix-community/nixops-gce/pull/9/commits/3ce6e4492140a55b5d22ebb1be26b063afd2055e